### PR TITLE
#22 mbean

### PR DESCRIPTION
**Pull request by krzysztofslusarski**

### CODE REVIEW

File: continuous-async-profiler-spring-starter/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerBootProperties.java [hunk 1]
```diff
@@ -74,18 +78,24 @@
      */
     private Pattern archiveCopyRegex = Pattern.compile(".*_13:0.*");
```
Comment (wyhasany) [inline continuous-async-profiler-spring-starter/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerBootProperties.java:90]: We can append here enum `properties-repository` to allow user easily choose between jmx/properties (there should appear autocompletion in editor properties)

File: continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanProperties.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+package com.github.krzysztofslusarski.asyncprofiler.mbean;
+
+import lombok.Value;
+import lombok.With;
+
+@With
+@Value
+class ContinuousAsyncProfilerMBeanProperties {
```
Comment (wyhasany) [inline continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanProperties.java]: What is the purpose of this file I don't see any usages of it?

File: continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanCondition.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+class ContinuousAsyncProfilerMBeanCondition implements Condition {
+    @Override
+    public boolean matches(ConditionContext conditionContext, AnnotatedTypeMetadata annotatedTypeMetadata) {
+        String repositoryType = conditionContext.getEnvironment().getProperty("async-profiler.continuous.properties-repository");
+        return "jmx".equalsIgnoreCase(repositoryType);
```
Comment (wyhasany) [inline continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanCondition.java:26]: I wonder if we should somehow respect Spring Boot default value of `spring.jmx.enabled=false`. What is your opinion?

File: continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerNotManageablePropertiesRepository.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.krzysztofslusarski.asyncprofiler;
+
+public interface ContinuousAsyncProfilerNotManageablePropertiesRepository {
+    ContinuousAsyncProfilerNotManageableProperties getAsyncProfilerNotManageableProperties();
```
Comment (wyhasany) [inline continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerNotManageablePropertiesRepository.java]: to be consistent with naming in ManagableProperties you could rename it to just `getNotManageableProperties()`

**Review by wyhasany — CHANGES_REQUESTED** [review]
There are a couple of little things to rethink :)

File: continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanCondition.java (hunk 3 above)
Comment (krzysztofslusarski) [reply continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/mbean/ContinuousAsyncProfilerMBeanCondition.java:26]: I thought about that but:
- I believe that it is only Spring Boot property
- I remember that I created JMX server on older Spring without that flag
- it is (unfortunately) common pattern, that libraries have their own "jmx" properties, eg: hikari

File: continuous-async-profiler-spring-starter/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerBootProperties.java [hunk 5]
```diff
@@ -57,6 +61,13 @@
         private Path archive = Paths.get("logs/archive");
     }
 
+    private PropertiesRepository propertiesRepository = PropertiesRepository.DEFAULT;
```
Comment (wyhasany) [inline continuous-async-profiler-spring-starter/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerBootProperties.java]: Consider add some documentation

**Review by wyhasany — APPROVED** [review]
Looks great!

**Pull request merged**